docs/contributing/sdk.md

```diff
@@ -0,0 +1,127 @@
+
+### 1. Set Up Development Environment
+
+Run the following command from the repository root directory:
+
+1. Create a new branch to hold contributed work:
+
+```bash
+❯ git checkout -b $BRANCH_NAME
+```
+
+2. Install all required dependencies:
+
+```bash
+❯ poetry install
+```
+
+3. Activate the virtual environment:
+
+```bash
+❯ poetry shell 
+```
+
+Run the following command to see where the new virtual environment is saved in the filesystem:
+
+```bash
+❯ poetry env info
+
+Virtualenv
+Python:         3.10.8
+Implementation: CPython
+Path:           /path/to/pypoetry/virtualenvs/demisto-sdk-zRo7lI35-py3.10
+Executable:     /path/to/pypoetry/virtualenvs/demisto-sdk-zRo7lI35-py3.10/bin/python
+Valid:          True
+```
+
+This output will be helpful when selecting the Python interpreter in [Visual Studio Code](https://code.visualstudio.com/docs/python/environments) and [PyCharm](https://www.jetbrains.com/help/pycharm/configuring-python-interpreter.html#add-existing-interpreter).
+
+You should now have a working development environment. Feel free to jump into your favorite IDE and open the repository.
+
+### 2. Add Work
+
+This is the step where you add new features, fix bugs, etc., commit and push them.
+
+To get you started, here's some useful information about the SDK project structure, important modules and how to create new commands. 
+#### Directory Structure
+
+The package that holds the source code for the commands, utilities and unit tests is `demisto_sdk`.
+
+The main module is located in `demisto_sdk/__main__.py` and it holds the business logic for initializing the SDK and parsing the commands/arguments. So to run the `demisto-sdk -h`, you could run the following command from within virtual environment:
+
+```bash
+❯ python demisto_sdk/__main__.py -h
+```
+
+Each command has it's own package under `demisto_sdk/commands`.
+
+#### Example Contribution: Create a New Command
+
+To create a new command, follow the steps below:
+
+1. Create package for your command in the `demisto_sdk/commands/$NEW_COMMAND` directory.
+2. Create the `click` command and arguments in the `__main__.py` module. See [Basic Concepts - Creating a Command](https://click.palletsprojects.com/en/8.1.x/quickstart/#basic-concepts-creating-a-command) for more information.
+3. Create a module in `demisto_sdk/commands/$NEW_COMMAND/$NEW_MODULE.py`.
+
+   **Note:** Modules are supposed to return `0` on success or `1` on failure. 
+1. Create unit tests. Unit tests should be saved in `demisto_sdk/commands/$NEW_COMMAND/tests`. Test files for all commands which located are located in 
+`demisto_sdk/tests/test_files`.
+   
+   To run the unit tests from within your virtual environment:
+
+   ```bash
+   ❯ pytest -v demisto_sdk/commands/$NEW_COMMAND/tests
+   ```
```

Comment:
it is also possible to run the tests in the IDE and debug them in case of issues, we should document it too